File: python/02631_IntrProg/week002/boxArea.py
def boxArea(boxCorners, area):
     
    # Extract coordinates
    x1 = boxCorners[0]
    x2 = boxCorners[1]
    x3 = boxCorners[2]
    x4 = boxCorners[3]
    y1 = boxCorners[4]
    y2 = boxCorners[5]
    y3 = boxCorners[6]
    y4 = boxCorners[7]
     
    # Compute area of box 1
    A1 = (x2-x1)*(y2-y1)
     
    # Compute area of box 2
    A2 = (x4-x3)*(y4-y3)
     
    # Compute area of overlap
    Ao = max(0, min(x2,x4)-max(x1,x3)) * max(0, min(y2,y4)-max(y1,y3))
     
    if area == "Box1":
        A = A1
    elif area == "Box2":
        A = A2
    elif area == "Intersection":
        A = Ao
    elif area == "Union":
        A = A1 + A2 - Ao       
     
    return A

File: MATLAB/02631_IntrProg/week002/boxArea.m
function A = boxArea(bxcr, area)
assert(length(bxcr)==8);
assert(min(bxcr)>=0)
assert(strcmp(area,'Box1') | strcmp(area,'Box2') | strcmp(area,'Intersection') | strcmp(area,'Union'));
% Use a switch statement to choose between the different areas to compute.
switch area
   case  'Box1'
      % compute area of box one
      A = (bxcr(2)-bxcr(1))*(bxcr(6)-bxcr(5));
      
   case  'Box2'
      % compute area of box two
      A = (bxcr(4)-bxcr(3))*(bxcr(8)-bxcr(7));
   case  'Intersection'
      % compute area of intersection
      A = max(0,min(bxcr(2),bxcr(4)) - max(bxcr(1),bxcr(3))) * max(0,min(bxcr(6),bxcr(8)) - max(bxcr(5),bxcr(7)));
   case  'Union'
      % Insert code to compute area of union
      A1 = (bxcr(2)-bxcr(1))*(bxcr(6)-bxcr(5));
      A2 = (bxcr(4)-bxcr(3))*(bxcr(8)-bxcr(7));
      A3 = max(0,min(bxcr(2),bxcr(4)) - max(bxcr(1),bxcr(3))) * max(0,min(bxcr(6),bxcr(8)) - max(bxcr(5),bxcr(7)));
      A = A1+A2-A3;
end